runtime/jcl/common/java_lang_invoke_VarHandle.c

```diff
@@ -111,7 +112,18 @@ Java_java_lang_invoke_FieldVarHandle_lookupField(JNIEnv *env, jobject handle, jc
 		goto _cleanup;
 	}
 
-	J9VMJAVALANGINVOKEVARHANDLE_SET_MODIFIERS(vmThread, J9_JNI_UNWRAP_REFERENCE(handle), ((J9ROMFieldShape*) romField)->modifiers);
+	varHandle = J9_JNI_UNWRAP_REFERENCE(handle);
+	J9VMJAVALANGINVOKEVARHANDLE_SET_MODIFIERS(vmThread, varHandle, ((J9ROMFieldShape *)romField)->modifiers);
+
+	/* StaticFieldVarHandle and InstanceFieldVarHandle extend FieldVarHandle. For StaticFieldVarHandle, isStatic
+	 * is true, and the definingClass field is set to the owner of the static field. For InstanceFieldVarHandle,
+	 * isStatic is false, and the definingClass field is set to the lookupClass, which is also the instance class,
+	 * in FieldVarHandle's constructor.
+	 */
+	if (isStatic && (j9LookupClass != definingClass)) {
```

Comment:
I would prefer this code unconditionally set the field for static handles.  This makes the code clearer and is safe for the future if the field is no longer set in java
```suggestion
	if (isStatic) {
```